fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -184,56 +187,108 @@ protected TemplateDruidQuery getMergedQuery(List<String> names) {
      * @param fieldName  The name for the aggregation or post aggregation column being gotten
      *
      * @return A post aggregator representing a number field value
+     *
+     * @deprecated use the static version {@link #getNumericField(MetricField)} by preference
      */
+    @Deprecated
     protected PostAggregation getNumericField(String fieldName) {
-        // Get the field
-        MetricField field = metrics.get(fieldName).getMetricField();
-
-        // Check for sketches, since we can't handle them in here
-        if (field.isSketch()) {
-            return FieldConverterSupplier.sketchConverter.asSketchEstimate(field);
-        }
-
-        // Handle it if it's an Aggregation (ie. wrap it in a fieldAccessorPostAggregation)
-        if (field instanceof Aggregation) {
-            field = new FieldAccessorPostAggregation((Aggregation) field);
-        }
+        return getNumericField(metrics.get(fieldName).getMetricField());
+    }
 
-        // If it's already a post-agg, we're good, and if it was an agg, we've already wrapped it
-        return (PostAggregation) field;
+    /**
+     * Prepare a post aggregation for a field expecting a numeric value.
+     * <p>
+     * The post-agg is created per the following heuristics:
+     * <dl>
+     *     <dt>If it's an aggregator
+     *     <dd>wrap it in a field accessor
+     *     <dt>If it's a sketch
+     *     <dd>wrap it in a sketch estimate
+     *     <dt>If it's already a numeric post aggregator
+     *     <dd>simply return it
+     * </dl>
+     *
+     * @param field  The field being coerced
+     *
+     * @return A post aggregator representing a number field value
+     */
+    protected static PostAggregation getNumericField(MetricField field) {
+        // If the field is a sketch, wrap it in a sketch estimate, if it's an aggregation, create a post aggregation
+        // Otherwise it is a number post aggregation already
```

Comment:
I agree that it is redundant.  I'll consider removing it.  I think it got duplicated up into the javadoc and wasn't pruned.